pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -267,23 +269,84 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		reqLogger.Info("Created Service", "Service", serDNS)
 
 		if infinispan.IsExposed() {
-			externalService := r.serviceExternal(infinispan)
-			if externalService != nil {
-				err = r.client.Create(context.TODO(), externalService)
-				if err != nil && !errors.IsAlreadyExists(err) {
-					reqLogger.Error(err, "failed to create external Service", "Service", ser)
+			switch infinispan.Spec.Expose.Type {
+			case infinispanv1.ExposeTypeLoadBalancer, infinispanv1.ExposeTypeNodePort:
+				externalService := r.serviceExternal(infinispan)
+				if externalService != nil {
+					err = r.client.Create(context.TODO(), externalService)
+					if err != nil && !errors.IsAlreadyExists(err) {
+						reqLogger.Error(err, "failed to create external Service", "Service", ser)
+						return reconcile.Result{}, err
+					}
+					err = r.client.Update(context.TODO(), infinispan)
+					if err != nil {
+						reqLogger.Error(err, "Failed to update Infinispan with Service", "Service", ser)
+						return reconcile.Result{}, err
+					}
+					reqLogger.Info("Created External Service", "Service", externalService)
+				} else {
+					reqLogger.Info("External Service NOT created. Unsupported type?", "Service", nil)
+
+				}
+			case infinispanv1.ExposeTypeRoute:
+				lsService := ispncom.LabelsResource(infinispan.ObjectMeta.Name, "infinispan-service-external")
+				route := openshiftv1.Route{
```

Comment:
@rigazilla what do you think about to check that we are running on OCP here? I'm not sure that on k8s this will print correct error during Route struct construction. We can implement isOpenshift flag in the ReconcileInfinispan and detect it at the early init phase.

Reply:
@dmvolod I was actually thinking to implement an automatic route/ingress selection like this:
https://github.com/application-stacks/runtime-component-operator/blob/d3bb9c7f29936169fd1e9092e255895980522ce1/pkg/controller/runtimecomponent/runtimecomponent_controller.go#L654

Reply:
how does it sound?

Reply:
Oh, @rigazilla , it would be nice to have this option. We can use just one expose type for this, like External or keep Route and remove Ingress and implement automatic choose as you suggested.

Reply:
@dmvolod should works as we discussed above now. I've used `Route` as generic exposition type

Reply:
Excellent implementation, thanks @rigazilla

Reply:
thank you!